File: credit.c
//Checks for valid Credit Card
//Outputs Credit Card Brand or Invalid

#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>

int main(void)
{
    //Only allows numbers between 13-16 digits
    long n;
    long m;
    long c;
    long b;
    long i = 10;
    int num = 0;
    int sum1 = 0;
    int sum2 = 0;
    int check;
    int checksum;
    int digits;
    int digit;

    do
    {
        n = get_long("Number: ");
        if (n > 9999999999999999 || n < 1000000000000)
        {
        printf("INVALID\n");
        exit(0);
        }
    }
    while (n > 9999999999999999 || n < 1000000000000);



    m = n;
    c = n;
    b = m;

    //Checksum Test
    //Checking Second Digit (from right)
    do
    {
        n = n / i;
        num = (n % 10) * 2;
        i = 100;

        if (num > 9)
        {
            num = (num % 10) + 1;
        }
        else
        {
            num = num + 0;
        }

        sum1 = sum1 + num;
        //printf("sum1=%i\n", sum1);

    }
    while (n > 0);

    //Checking First Digit (from right)
    do
    {
        sum2 = sum2 + (m % 10);
        m = m / i;
    }
    while (m > 0);

    //printf("sum1=%i\n", sum1);
    //printf("sum2=%i\n", sum2);

    //Checksum Test
    check = sum1 + sum2;
    checksum = check % 10;

    printf("checksum=%i\n", checksum);

    //Finding First Digit, Two Digits on the card
    do
    {
        c = c / 10;
        digits = c % 100;
    }
    while (c > 100);

    do
    {
        b = b / 10;
        digit = b % 10;
    }
    while (b > 10);

    //Outputting Brand of card

    if ((digits == 34 || digits == 37) && checksum == 0)
    {
        printf("AMEX\n");
    }
    else if (digit == 4 && checksum == 0)
    {
        printf("VISA\n");
    }
    else if ((digits == 51 || digits == 52 || digits == 53 || digits == 54 || digits == 55) && checksum == 0)
    {
        printf("MASTERCARD\n");
    }
    else
    {
        printf("INVALID\n");
    }

}

File: credit.py
from cs50 import get_int
import math

# prompts user to enter credit card number
# checks for valid entry

while True:
    n = get_int("Number: ")

    if n > 9999999999999999 or n < 1000000000000:
        print("INVALID")
        exit(0)

    if n > 1000000000000 and n < 9999999999999999:
        break

m = n
c = n
b = m
i = 10
sum1 = 0
sum2 = 0

# Checksum Test
# Checking Second Digit (from right)
while n > 0:
    n = math.floor(n / i)
    num = math.floor((n % 10) * 2)
    i = 100

    if num > 9:
        num = math.floor((num % 10) + 1)

    else:
        num = num + 0

    sum1 += num

# Checking First Digits (from right)
while m > 0:
    sum2 += math.floor(m % 10)
    m = m / i

# Checksum Test
checksum = (sum1 + sum2) % 10

while c > 100:
    c = c / 10
    digits = math.floor(c % 100)

while b > 10:
    b = b / 10
    digit = math.floor(b % 10)

# Outputting the Brand of the Card
if (digits == 34 or digits == 37) and checksum == 0:
    print("AMEX")

elif digit == 4 and checksum == 0:
    print("VISA")

elif (digits == 51 or digits == 52 or digits == 53 or digits == 54 or digits == 55) and checksum == 0:
    print("MASTERCARD")

else:
    print("INVALID")